src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -105,6 +108,8 @@ public function collect(Request $request, Response $response, \Throwable $except
             'response_cookies' => $responseCookies,
             'session_metadata' => $sessionMetadata,
             'session_attributes' => $sessionAttributes,
+            'session_usages' => array_values($this->sessionUsages),
+            'stateless_check' => $request->attributes->get('_stateless'),
```

Comment:
this is not how the AbstractSessionListener checks this attribute, when sub-requests are involved.

Reply:
You're completely right, thanks!
I updated the code in order to use the request stack instead (and I added related tests as well)